TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+            set;
+        }
+
+        /// <summary>
+        /// The current Sentence in the current Paragraph.
+        /// </summary>
+        internal CfgSentence CurrentSentence
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The Section and Paragraph Domain of this program.
+        /// </summary>
+        internal Container<Symbol> SectionsParagraphs
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Ordered list of all Sections an Paragraphs encountered in order.
+        /// </summary>
+        internal List<Symbol> AllSectionsParagraphs
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// All pending Goto instructions that will be handled at the end of the Procedure Division.
+        /// </summary>
+        protected LinkedList<Tuple<Goto, BasicBlockForNode>> PendingGOTOs;
+        /// <summary>
+        /// All encountered PERFORM procedure instructions
+        /// </summary>
+        protected LinkedList<Tuple<PerformProcedure, BasicBlockForNodeGroup>> PendingPERFORMProcedures;
+        /// <summary>
+        /// Pending ALTER instructions that will be handled at the end of the Procedure Division.
+        /// </summary>
+        protected LinkedList<Tuple<Alter, BasicBlockForNode>> PendingALTERs;
+
+        /// <summary>
+        /// More Symbol Reference associated to altered GOTO statement.
+        /// </summary>
+        protected Dictionary<Goto, HashSet<SymbolReference>> PendingAlteredGOTOS;
+
+        /// <summary>
+        /// All pending Next Sentence instructions that will be handled at the end of the Procedure Division.
+        /// </summary>
+        internal LinkedList<Tuple<NextSentence, BasicBlockForNode, CfgSentence>> PendingNextSentences;
+
+        /// <summary>
+        /// All encountered sentences
+        /// </summary>
+        internal List<CfgSentence> AllSentences;
+
+        public IList<Diagnostic> Diagnostics { get; private set; }
+        /// <summary>
+        /// Empty constructor.
+        /// </summary>
+        public ControlFlowGraphBuilder() : this(null)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="parentCfgBuilder">Parent Control Flow Builder for a nested program</param>
+        public ControlFlowGraphBuilder(ControlFlowGraphBuilder<D> parentCfgBuilder = null)
+        {
+            this.ParentProgramCfgBuilder = parentCfgBuilder;
+            this.Diagnostics = new List<Diagnostic>();
+            this.UseEvaluateCascade = true;
+            this.UseSearchCascade = true;
+            Mode = CfgMode.Normal;
+        }
+
+        /// <summary>
+        /// Called when a Node has been completely parsed.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="program"></param>
+        public override void OnNode(Node node, Program program)
```

Comment:
Remark for the entire class: remove empty overrides.

Reply:
Yes, cleaning code

Reply:
done